internal/controllers/machine/drain/drain.go

```diff
@@ -0,0 +1,353 @@
+			log.Error(err, "Error when evicting Pod")
+			res.PodsFailedEviction[err.Error()] = append(res.PodsFailedEviction[err.Error()], pd.Pod)
+		case apierrors.IsForbidden(err) && apierrors.HasStatusCause(err, corev1.NamespaceTerminatingCause):
+			// Creating an eviction resource in a terminating namespace will throw a forbidden error, e.g.:
+			// "pods "pod-6-to-trigger-eviction-namespace-terminating" is forbidden: unable to create new content in namespace test-namespace because it is being terminated"
+			// The kube-controller-manager is supposed to set the deletionTimestamp on the Pod and then this error will go away.
+			msg := "Cannot evict pod from terminating namespace: unable to create eviction (kube-controller-manager should set deletionTimestamp)"
+			log.Error(err, msg)
+			res.PodsFailedEviction[msg] = append(res.PodsFailedEviction[msg], pd.Pod)
+		default:
+			log.Error(err, "Error when evicting Pod")
+			res.PodsFailedEviction[err.Error()] = append(res.PodsFailedEviction[err.Error()], pd.Pod)
+		}
+	}
+
+	return res
+}
+
+// evictPod evicts the given Pod, or return an error if it couldn't.
+func (d *Helper) evictPod(ctx context.Context, pod *corev1.Pod) error {
+	delOpts := metav1.DeleteOptions{}
+	if d.GracePeriodSeconds >= 0 {
+		gracePeriodSeconds := int64(d.GracePeriodSeconds)
+		delOpts.GracePeriodSeconds = &gracePeriodSeconds
+	}
+
+	eviction := &policyv1.Eviction{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pod.Name,
+			Namespace: pod.Namespace,
+		},
+		DeleteOptions: &delOpts,
+	}
+
+	return d.Client.SubResource("eviction").Create(ctx, pod, eviction)
+}
+
+// EvictionResult contains the results of an eviction.
+type EvictionResult struct {
+	PodsDeletionTimestampSet []*corev1.Pod
+	PodsFailedEviction       map[string][]*corev1.Pod
+	PodsNotFound             []*corev1.Pod
+	PodsIgnored              []*corev1.Pod
+}
+
+// DrainCompleted returns if a Node is entirely drained, i.e. if all relevant Pods have gone away.
+func (r EvictionResult) DrainCompleted() bool {
+	return len(r.PodsDeletionTimestampSet) == 0 && len(r.PodsFailedEviction) == 0
+}
+
+// ConditionMessage returns a condition message for the case where a drain is not completed.
+func (r EvictionResult) ConditionMessage() string {
+	if r.DrainCompleted() {
+		return ""
+	}
+
+	conditionMessage := "Drain not completed yet:"
+	if len(r.PodsDeletionTimestampSet) > 0 {
+		conditionMessage = fmt.Sprintf("%s\n* Pods with deletionTimestamp that still exist: %s", conditionMessage, stringPodList(r.PodsDeletionTimestampSet, 5))
+	}
+	if len(r.PodsFailedEviction) > 0 {
+		sortedFailureMessages := maps.Keys(r.PodsFailedEviction)
+		sort.Strings(sortedFailureMessages)
+
+		if len(sortedFailureMessages) <= 5 {
+			conditionMessage = fmt.Sprintf("%s\n* Pods with eviction failed:", conditionMessage)
+		} else {
+			sortedFailureMessages = sortedFailureMessages[:5]
+			conditionMessage = fmt.Sprintf("%s\n* Pods with eviction failed: (showing only first 5)", conditionMessage)
```

Comment:
```suggestion
			conditionMessage = fmt.Sprintf("%s\n* Pods with eviction failed: (showing only first 5 of %d)", conditionMessage, len(sortedFailureMessages))
```

Reply:
Getting back to this. 
We have failure messages, and for each failure message a list of pods to which the error applies.

We are showing up to 5 messages, and max 3 pod each.
That's ok

However, it should be great to give infos about how many other pods failed to drain.

eg.

message1
```
* Pods with deletionTimestamp that still exist: pod-2-deletionTimestamp-set-1, pod-3-to-trigger-eviction-successfully-1 ... (5 other pods)
* Pods with eviction failed:
  * Cannot evict pod as it would violate the pod's disruption budget. The disruption budget pod-5-pdb needs 20 healthy pods and has 20 currently: pod-5-to-trigger-eviction-pdb-violated-1 
  * some other error 1: pod-6-to-trigger-eviction-some-other-error`
  * ... (10 other errors applying to 15 pods)
```

Reply:
That's fine, good idea! Will do